### ISSUE TITLE
Improve auto type (List or Object) for CSV to JSON conversion

### ISSUE THREAD

**Issue by tuanchauict** [theapache64/retrosheet#11]
In `RetrosheetInterceptor.isReturnTypeList()`, I see we are using `contains("java.util.List")` to check whether the return type is in list type or not. This limits auto type to just List but not the other collection type like Set, Array.

Besides, suspend function does not return a list but a `kotlin.coroutines.Continuation` param is appended at the last of the parameter list. It seems to me that the code cannot check suspend function.

Using `contains` also dangerous because if a generic type (not collection) accepts `List` as the type param, `contains` will cause incorrect. For example:
```
fun foo(): Foo<Bar<List<String>>>
```
One more problem (although we don't expect to use retrosheet for a real product), obfuscation may rename List to anything..

I suggest a new check like this:

```kt
private fun isSuspendMethodWithListReturnValue(method: Method): Boolean {
    val lastParameter = method.genericParameterTypes.lastOrNull() ?: return false
    val typeName = lastParameter.typeName
    val continuationClassName =
        kotlin.coroutines.Continuation::class.qualifiedName ?: return false
    val isSuspendFun = typeName.startsWith("$continuationClassName<")
    if (!isSuspendFun) {
        return false
    }
    // TODO: We can shorten this with Regex
    val suspendFunReturnType = typeName
        .removePrefix("$continuationClassName<")
        .split("<")
        .first()
        .split(" ")
        .last()
    val clazz = Class.forName(suspendFunReturnType)
    return Collection::class.java.isAssignableFrom(clazz) || clazz.isArray
}
```

**Comment by tuanchauict** [comment]
It also seems to me that `Method::class.java.getDeclaredField("signature")` works on JVM but not on Android.
I could run it for Unit test but cannot run on Android app.

**Comment by theapache64** [comment]
You're right. To fix this issue, I've recently created a new annotation called `@ReadAsList`. But I think this is much better and avoids the usage of extra annotation.

**Comment by theapache64** [comment]
Will be available from `2.0.0-alpha05`. 

@tuanchauict Thank you so much for this solution :smile:
